src/scene/vcPOI.cpp

```diff
@@ -1344,19 +1344,14 @@ void vcPOI::GenerateLineFillPolygon(vcState *pProgramState)
     
     // Calculate area
     m_meshArea = 0.0;
-    for (int i = 0; i < trianglePointList.size(); i += 3)
+    for (size_t i = 0; i < trianglePointList.size(); i += 3)
     {
-      udFloat3 e1 = {
-        pVerts[i + 1].position.x - pVerts[i].position.x,
-        pVerts[i + 1].position.y - pVerts[i].position.y,
-        pVerts[i + 1].position.z - pVerts[i].position.z };
-      udFloat3 e2 = {
-        pVerts[i + 2].position.x - pVerts[i].position.x,
-        pVerts[i + 2].position.y - pVerts[i].position.y,
-        pVerts[i + 2].position.z - pVerts[i].position.z };
-      udFloat3 e3 = udCross3<float>(e1, e2);
-
-      m_meshArea += (double)(udMag3(e3) / 2.0f);
+      udDouble3 triPoint1 = udDouble3::create(pVerts[i].position);
```

Comment:
I guess I mean to say; it looks like `pVerts` is created from `trianglePointList`. `pVerts` looks to be floats, so we lose precision when we convert from double to float. Can we calculate the area using `trianglePointList` instead? Then we can keep double precision all the way through.

Reply:
Ok i've fixed it so that it does all of the math in doubles until the end where it converts it to float for the final render